engine/src/main/java/io/graphqlcrud/DatabaseSchemaBuilder.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright 2012-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.graphqlcrud;
+
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.graphqlcrud.model.Attribute;
+import io.graphqlcrud.model.Cardinality;
+import io.graphqlcrud.model.Entity;
+import io.graphqlcrud.model.Relation;
+import io.graphqlcrud.model.Schema;
+
+public class DatabaseSchemaBuilder {
+    private static final Logger LOGGER = LoggerFactory.getLogger(DatabaseSchemaBuilder.class);
+    private final DatabaseMetaData databaseMetaData;
+    private final String schema;
+
+    protected DatabaseSchemaBuilder(Connection connection, String schema) throws SQLException {
+        this.databaseMetaData = connection.getMetaData();
+        this.schema = schema;
+    }
+
+    public static Schema getSchema(Connection connection, String schema) throws SQLException {
+        DatabaseSchemaBuilder analyzer = new DatabaseSchemaBuilder(connection, schema);
+        return analyzer.buildSchema();
+    }
+
+    protected Schema buildSchema() throws SQLException {
+        Schema s = new Schema(this.schema);
+        Map<String, Entity> entityMap = new TreeMap<String, Entity>();
+        try (ResultSet results = this.databaseMetaData.getTables(null, this.schema, null,
+                new String[] { "TABLE", "VIEW" })) {
+            while (results.next()) {
+                String tableName = results.getString("TABLE_NAME");
+                if (tableName.equals("flyway_schema_history")) {
```

Comment:
I would add TODO for user-provided list of include/exclude patterns

Reply:
Yeap.